src/mds/PurgeQueue.cc

```diff
@@ -594,7 +594,7 @@ void PurgeQueue::_execute_item(
 
   in_flight[expire_to] = item;
   logger->set(l_pq_executing, in_flight.size());
-  files_high_water = std::max(files_high_water,
+  files_high_water = std::max(static_cast<uint64_t>(files_high_water),
```

Comment:
why? `files_high_water` is already an `uint64_t`.

Reply:
i just read the description in the tracker. but i still don't understand why GCC fails to figure out the first parameter is actually an `uint64_t`. probably a less confusing change is to have something like
```c++
files_high_water = std::max<uint64_t>(files_high_water,
                                      in_flight.size());
``` 

but still it's weird..

Reply:
Log snippet
`no matching function for call to 'max(size_t&, uint64_t)'`

it seems to assigned size_t by gcc?

Reply:
Complete failed build log here:
https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/387736/raw

Reply:
just read it. assuming you are using v16.2.3, i don't have a clue explaining this. probably it's a GCC bug. but i cannot find any bug report similar to this one
```diff
diff --git a/src/mds/PurgeQueue.h b/src/mds/PurgeQueue.h
index 6e953d3e633..dd678de4088 100644
--- a/src/mds/PurgeQueue.h
+++ b/src/mds/PurgeQueue.h
@@ -245,5 +245,6 @@ private:

   uint64_t ops_high_water = 0;
   uint64_t files_high_water = 0;
+  static_assert(std::is_same_v<decltype(files_high_water), uint64_t>);
 };
 #endif
```
if you are able to reproduce this locally. probably you could give the patch above a try.

Reply:
Yes using 16.2.3.  I can try on a pi4 and see if it compiles

Reply:
FWIW, #41239 is created to address the warnings in https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/387736/raw.

Reply:
Thanks will add that patch as well